src/control/server/mgmt_svc.go

```diff
@@ -470,6 +470,59 @@ func (svc *mgmtSvc) smdReplace(ctx context.Context, req *mgmtpb.SmdQueryReq) (*m
 	}, nil
 }
 
+func (svc *mgmtSvc) smdIdentify(ctx context.Context, req *mgmtpb.SmdQueryReq) (*mgmtpb.SmdQueryResp, error) {
+	req.Rank = uint32(system.NilRank)
+	rank, device, err := svc.smdQueryDevice(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	if device == nil {
+		return nil, errors.Errorf("smdIdentify on %s did not match any devices", req.Uuid)
+	}
+
+	// TODO: Make sure the device is a VMD device (Here??)
```

Comment:
TODO/FIXME comments should ideally be associated with a ticket, e.g. "TODO (DAOS-XXX): Make sure ...". That way reviewers know that there is a plan to address whatever the comment is describing.

I'm a bit concerned about landing this with minimal test coverage (there are no unit tests for the RPC handler) and no validation for the device type. What happens if you run this on a non-VMD device? Does it just not work or is there potentially bad behavior?

Reply:
@mjmac so currently if a non VMD device is used, the command will fail with a DER_NOSYS error, and in the server logs it will have the ERR in BIO indicated that the device is not a VMD device. I will file a ticket to add testing for this command and do the necessary cleanup, and also tag that in the comment (although I might remove the comment as it might not be necessary to fail here if non-VMD device)